### ISSUE TITLE
Data race during startup

### ISSUE THREAD

**Issue by mhmxs** [opencost/opencost#2827]
**Describe the bug**
I started a debug session with `-race` flag, and it catches a data race during startup:

```
==================
WARNING: DATA RACE
Write at 0x00c003ef71d0 by goroutine 369:
  runtime.mapassign_faststr()
      /nix/store/4wsyskr8rgrkq5ahkkwnxbrg4p1nlnfb-go-1.22.4/share/go/src/runtime/map_faststr.go:203 +0x0
  github.com/opencost/opencost/pkg/costmodel.(*CostModel).ComputeCostData()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/costmodel.go:396 +0x7a2c
  github.com/opencost/opencost/pkg/costmodel.(*CostModelMetricsEmitter).Start.func1()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/metrics.go:457 +0x176b

Previous read at 0x00c003ef71d0 by goroutine 368:
  runtime.mapaccess2_faststr()
      /nix/store/4wsyskr8rgrkq5ahkkwnxbrg4p1nlnfb-go-1.22.4/share/go/src/runtime/map_faststr.go:108 +0x0
  k8s.io/apimachinery/pkg/labels.Set.Has()
      /home/rkovacs/.go/1_22/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/labels/labels.go:53 +0x84
  k8s.io/apimachinery/pkg/labels.(*Requirement).Matches()
      /home/rkovacs/.go/1_22/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/labels/selector.go:226 +0x381
  k8s.io/apimachinery/pkg/labels.internalSelector.Matches()
      /home/rkovacs/.go/1_22/pkg/mod/k8s.io/apimachinery@v0.30.2/pkg/labels/selector.go:387 +0xa9
  k8s.io/apimachinery/pkg/labels.(*internalSelector).Matches()
      <autogenerated>:1 +0xa4
  github.com/opencost/opencost/pkg/costmodel.getPodServices()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/costmodel.go:1428 +0x6e4
  github.com/opencost/opencost/pkg/costmodel.(*CostModel).costDataRange()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/costmodel.go:1794 +0x6ed3
  github.com/opencost/opencost/pkg/costmodel.(*CostModel).ComputeCostDataRange.func1()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/costmodel.go:1689 +0x228
  golang.org/x/sync/singleflight.(*Group).doCall.func2()
      /home/rkovacs/.go/1_22/pkg/mod/golang.org/x/sync@v0.7.0/singleflight/singleflight.go:198 +0x182
  golang.org/x/sync/singleflight.(*Group).doCall()
      /home/rkovacs/.go/1_22/pkg/mod/golang.org/x/sync@v0.7.0/singleflight/singleflight.go:200 +0x308
  golang.org/x/sync/singleflight.(*Group).Do()
      /home/rkovacs/.go/1_22/pkg/mod/golang.org/x/sync@v0.7.0/singleflight/singleflight.go:113 +0x679
  github.com/opencost/opencost/pkg/costmodel.(*CostModel).ComputeCostDataRange()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/costmodel.go:1688 +0x70d
  github.com/opencost/opencost/pkg/costmodel.(*Accesses).ComputeAggregateCostModel()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/aggregation.go:1415 +0x4011
  github.com/opencost/opencost/pkg/costmodel.(*Accesses).warmAggregateCostModelCache.func1()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/aggregation.go:1785 +0x1464
  github.com/opencost/opencost/pkg/costmodel.(*Accesses).warmAggregateCostModelCache.func2()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/aggregation.go:1818 +0xee
  github.com/opencost/opencost/pkg/costmodel.(*Accesses).warmAggregateCostModelCache.gowrap1()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/aggregation.go:1824 +0x46

Goroutine 369 (running) created at:
  github.com/opencost/opencost/pkg/costmodel.(*CostModelMetricsEmitter).Start()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/metrics.go:403 +0x111
  github.com/opencost/opencost/pkg/costmodel.Initialize()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/router.go:1730 +0x55c7
  github.com/opencost/opencost/pkg/cmd/costmodel.Execute()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/cmd/costmodel/costmodel.go:46 +0x313
  github.com/opencost/opencost/pkg/cmd.newCostModelCommand.func1()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/cmd/commands.go:108 +0x5c
  github.com/spf13/cobra.(*Command).execute()
      /home/rkovacs/.go/1_22/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:856 +0x13b7
  github.com/spf13/cobra.(*Command).ExecuteC()
      /home/rkovacs/.go/1_22/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0xe55
  github.com/spf13/cobra.(*Command).Execute()
      /home/rkovacs/.go/1_22/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902 +0x4b
  github.com/opencost/opencost/pkg/cmd.Execute()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/cmd/commands.go:61 +0x7d2
  main.main()
      /home/rkovacs/src/github.com/opencost/opencost/cmd/costmodel/main.go:11 +0x4d

Goroutine 368 (running) created at:
  github.com/opencost/opencost/pkg/costmodel.(*Accesses).warmAggregateCostModelCache()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/aggregation.go:1810 +0x35a
  github.com/opencost/opencost/pkg/costmodel.Initialize()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/costmodel/router.go:1724 +0x5526
  github.com/opencost/opencost/pkg/cmd/costmodel.Execute()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/cmd/costmodel/costmodel.go:46 +0x313
  github.com/opencost/opencost/pkg/cmd.newCostModelCommand.func1()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/cmd/commands.go:108 +0x5c
  github.com/spf13/cobra.(*Command).execute()
      /home/rkovacs/.go/1_22/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:856 +0x13b7
  github.com/spf13/cobra.(*Command).ExecuteC()
      /home/rkovacs/.go/1_22/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0xe55
  github.com/spf13/cobra.(*Command).Execute()
      /home/rkovacs/.go/1_22/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902 +0x4b
  github.com/opencost/opencost/pkg/cmd.Execute()
      /home/rkovacs/src/github.com/opencost/opencost/pkg/cmd/commands.go:61 +0x7d2
  main.main()
      /home/rkovacs/src/github.com/opencost/opencost/cmd/costmodel/main.go:11 +0x4d
==================
```

**To Reproduce**
1. Compile binary with `-race` flag
2. Start application

**Expected behavior**
Data race never should happen

**Screenshots**

**Which version of OpenCost are you using?**
`develop`

**Additional context**
Add any other context about the problem here. Kubernetes versions and which public clouds you are working with are especially important.

**Comment by mattray** [comment]
Thanks, I've escalated this to Kubecost Engineering.

**Comment by AjayTripathy** [comment]
Hi @mhmxs I believe we have this fixed in 2.3.3 . Can you take a look at that version?

**Comment by mhmxs** [comment]
> Hi @mhmxs I believe we have this fixed in 2.3.3 . Can you take a look at that version?

It has gone :pray:
